src/main/java/us/kilroyrobotics/subsystems/Wrist.java

```diff
@@ -5,19 +5,107 @@
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import java.util.function.Supplier;
+import us.kilroyrobotics.Constants.CoralMechanismConstants;
+import us.kilroyrobotics.Constants.SimulationConstants;
 
 public class Wrist extends SubsystemBase {
+    private SparkMax m_wristMotor;
     private Supplier<Pose3d> getCarriagePose;
+    private boolean m_useAbsoluteEncoder;
+    private SparkAbsoluteEncoder m_absoluteEncoder;
+    private RelativeEncoder m_relativeEncoder;
+    private SparkClosedLoopController m_pidController;
+
+    /* Sim Specific */
+    private DCMotor m_simWristGearbox;
+    private SparkMaxSim m_simWristMotor;
+    private SingleJointedArmSim m_simWrist;
 
     /** Creates a new Wrist. */
-    public Wrist(Supplier<Pose3d> carriagePoseGetter) {
+    public Wrist(Supplier<Pose3d> carriagePoseGetter, boolean useAbsoluteEncoder) {
+        this.m_wristMotor =
+                new SparkMax(CoralMechanismConstants.kWristMotorId, MotorType.kBrushless);
+        this.m_pidController = this.m_wristMotor.getClosedLoopController();
+        this.m_useAbsoluteEncoder = useAbsoluteEncoder;
+        if (useAbsoluteEncoder) {
+            this.m_absoluteEncoder = this.m_wristMotor.getAbsoluteEncoder();
+        } else {
+            this.m_relativeEncoder = this.m_wristMotor.getEncoder();
+        }
+
+        // Configure
+        SparkMaxConfig wristMotorConfig = new SparkMaxConfig();
+        wristMotorConfig
+                .closedLoop
+                .feedbackSensor(
+                        useAbsoluteEncoder
+                                ? FeedbackSensor.kAbsoluteEncoder
+                                : FeedbackSensor.kPrimaryEncoder)
+                .pidf(
+                        CoralMechanismConstants.kP,
+                        CoralMechanismConstants.kI,
+                        CoralMechanismConstants.kD,
+                        CoralMechanismConstants.kF);
+        wristMotorConfig.idleMode(IdleMode.kBrake);
+        wristMotorConfig.smartCurrentLimit(40);
+
+        this.m_wristMotor.configure(
+                wristMotorConfig, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+
         this.getCarriagePose = carriagePoseGetter;
+
+        // Sim
+        this.m_simWristGearbox = DCMotor.getNEO(1);
+        this.m_simWristMotor = new SparkMaxSim(this.m_wristMotor, m_simWristGearbox);
+        this.m_simWrist =
+                new SingleJointedArmSim(
+                        m_simWristGearbox,
+                        SimulationConstants.kWristGearing,
+                        SingleJointedArmSim.estimateMOI(
+                                SimulationConstants.kArmLength.magnitude(),
+                                SimulationConstants.kWristMass.magnitude()),
+                        SimulationConstants.kArmLength.magnitude(),
+                        SimulationConstants.kMinAngle.in(Radians),
+                        SimulationConstants.kMaxAngle.in(Radians) * (Math.PI * 2),
```

Comment:
Remove the parentheses, they're unnecessary.

Also, should probably add a comment explaining why this needs done

Reply:
Will do